### DISCUSSION TITLE
How can I get the 'predict' result in real time?

### DISCUSSION THREAD

**Discussion by ujumjum** [ultralytics/ultralytics#5976] [Q&A]
I want to use the model.predict - source = 0(webcam) in real time.
And sending the label names of predict results in real time using the MQTT..

How can I get the predict result in real time??
YOLOv8

**Comment by pderrenger** [comment]
@ujumjum to obtain real-time predictions from a webcam using YOLOv8 and send the label names using MQTT, you can use the `predict` mode with `source=0` to access the webcam. You'll need to set up an MQTT client to publish the results. Here's a concise example of how you might set this up in Python:

```python
from ultralytics import YOLO
import paho.mqtt.client as mqtt

# Initialize YOLOv8 model
model = YOLO('yolov8n.pt')  # Load your pretrained model

# Setup MQTT client and connect
mqtt_client = mqtt.Client()
mqtt_client.connect("MQTT_BROKER_ADDRESS", MQTT_BROKER_PORT, 60)

# Start real-time prediction
results = model.predict(source=0, stream=True)  # Use stream=True for real-time

for result in results:
    # Extract label names from the prediction results
    labels = [model.names[int(cls)] for cls in result.boxes.cls]

    # Publish label names using MQTT
    for label in labels:
        mqtt_client.publish("TOPIC_NAME", label)

    # Handle other real-time display or processing needs here
    # ...

# Don't forget to disconnect the MQTT client when done
mqtt_client.disconnect()
```

Make sure to replace `MQTT_BROKER_ADDRESS`, `MQTT_BROKER_PORT`, and `TOPIC_NAME` with your MQTT broker's address, port, and the topic you wish to publish to, respectively.

This script will continuously get predictions from the webcam and send the detected class labels to the specified MQTT topic in real-time. You can add additional logic as needed for your specific use case.
